app/middlewares/generics.js

```diff
@@ -1,8 +1,25 @@
-const logger = require('../logger');
+const logger = require('../logger'),
+  jwt = require('jsonwebtoken'),
+  errors = require('../errors'),
+  config = require('../../config');
 
 exports.logRequestInformation = (req, res, next) => {
   logger.info(`A new request received at ${new Date()}`);
   logger.info('Request URL:', req.originalUrl);
   logger.info('Request Type:', req.method);
   next();
 };
+
+exports.verifyToken = (req, res, next) => {
+  const token = req.body.token || req.query.token || req.headers[config.common.session.header_name];
+
+  if (token) {
+    return jwt.verify(token, config.common.session.secret, (err, decoded) => {
```

Comment:
didnt you already change this with a promisified version?

Reply:
Not here. Why should I do that? I made it fail on purpose and the error handling was ok.